### PR TITLE
Iceflow with icepyx notebook

### LINKED ISSUE #29: Develop example showing `iceflow` and `icepyx` working together
Integrate the usage of [icepyx](https://icepyx.readthedocs.io/en/latest/) to allow search/download & harmonization of ICESat2 data.

### PR DESCRIPTION
**Pull request by trey-stafford**
Resolves #29 

----
📚 Documentation preview 📚: https://iceflow--50.org.readthedocs.build/en/50/

### CODE REVIEW

**Review by asteiker — CHANGES_REQUESTED** [review]
I will commit updates I made while I was reviewing. For iceflow-example, I cleared the code outputs because I ran into errors with my install. For iceflow-with-icepyx I tried not to actually run the cells but I accidentally did on atl06 resampling block. 

Overall these notebooks are excellent and most of my comments, other than the install issues, are just minor suggestions to improve the flow and written guidance.

File: docs/iceflow-example.ipynb
[the remark this answers is not in the document]
Comment (trey-stafford) [reply docs/iceflow-example.ipynb:1]: This is a good observation, I created an issue here to address: https://github.com/nsidc/iceflow/issues/51

File: notebooks/iceflow-example.ipynb
[the remark this answers is not in the document]
Comment (trey-stafford) [reply notebooks/iceflow-example.ipynb:1]: Yes, this is a bit confusing and could use some documentation. I setup the `notebooks/` directory for development, and have a script there that renders the notebooks for the docs (which get placed in the `docs/` dir). 

The `notebooks/` directory contains the source-of-truth, and is what we should make changes against. 

I know you made some changes to the rendered `docs/` notebooks, which is not a problem! I will transfer those changes to the `notebooks/` copies and re-render.

File: notebooks/iceflow-with-icepyx.ipynb
[the remark this answers is not in the document]
Comment (trey-stafford) [reply notebooks/iceflow-with-icepyx.ipynb:1]: Yes, see my comment in https://github.com/nsidc/iceflow/pull/50#discussion_r1834828618

File: notebooks/iceflow-example.ipynb
[the remark this answers is not in the document]
Comment (trey-stafford) [reply notebooks/iceflow-example.ipynb:1]: fe91a15

[the remark this answers is not in the document]
Comment (trey-stafford) [reply notebooks/iceflow-example.ipynb:1]: I also updated the `CONTRIBUTING.md` to include some additional information on developing jupyter notebooks.

File: docs/iceflow-with-icepyx.ipynb
[the remark this answers is not in the document]
Comment (trey-stafford) [reply docs/iceflow-with-icepyx.ipynb:1]: I updated the notebooks to use `df.head()` wherever we preview dataframes. Looks much more compact! 35dcc10

File: docs/iceflow-example.ipynb
[the remark this answers is not in the document]
Comment (trey-stafford) [reply docs/iceflow-example.ipynb:1]: I updated the notebooks to use `df.head()` wherever we preview dataframes. Looks much more compact! 35dcc10

[the remark this answers is not in the document]
Comment (trey-stafford) [reply docs/iceflow-example.ipynb:1]: I added a note to the beginning of the example notebook about the `iceflow-with-icepyx` notebook:

> To learn about how to download and manage larger amounts of data across many datasets with nsidc-iceflow, see the [Using iceflow with icepyx to Generate an Elevation Timeseries]() notebook.

dd75b99

I'm also going to add some higher-level docs about the library and supported datasets. That seems like a gap right now.

[the remark this answers is not in the document]
Comment (trey-stafford) [reply docs/iceflow-example.ipynb:1]: Improvements added in #53

**Pull request merged**